django/contrib/admin/helpers.py

```diff
@@ -279,6 +279,7 @@ def fields(self):
                 continue
             if not self.has_change_permission or field_name in self.readonly_fields:
                 yield {
+                    'name': field_name,
```

Comment:
Could you add a test for this change?

Reply:
Yes, will do.

Reply:
The new test covers the other change (which was already tested by the other test changes) but I still don't see any failures with this line reverted.

Reply:
Indeed, I must've overlooked that. The test has been fixed for this specific change.